internal/cmd/gocdk/apply_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"testing"
+)
+
+func TestApply(t *testing.T) {
+	// TODO(cla): test cases
+	// didn't supply biome name to command
+	// named biome doesn't exist
+	// no terraform file
+	// terraform not initialized *
+	// terraform never previously applied
+	// terraform apply repeat
+
+	if _, err := exec.LookPath("terraform"); err != nil {
+		t.Skip("terraform not found:", err)
+	}
+
+	t.Run("TerraformNotInitialized", func(t *testing.T) {
+		dir, cleanup, err := newTestModule()
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer cleanup()
+		const biomeName = "dev"
+		const wantGreeting = "HALLO WORLD"
+		if err := newTestBiome(dir, biomeName, wantGreeting); err != nil {
+			t.Fatal(err)
+		}
+		pctx := &processContext{
+			workdir: dir,
+			stdout:  ioutil.Discard,
+			stderr:  ioutil.Discard,
+		}
+		ctx := context.Background()
+
+		if err := run(ctx, pctx, []string{"apply", biomeName}, new(bool)); err != nil {
+			t.Errorf("run error: %+v", err)
+		}
+
+		outputs, err := tfReadOutput(findBiomeDir(dir, biomeName))
```

Comment:
Let's add a comment above this line explaining the assertion we're making: after a successful `terraform apply`, `terraform output` should return the greeting we configured. One non-obvious property (we ended up manually testing) was that `terraform output` fails if `terraform init` was not called, so this is actually verifying quite a bit of state with a seemingly simple assertion.